ui/app/controllers/allocations/allocation/index.js

```diff
@@ -29,6 +29,30 @@ export default Controller.extend(Sortable, {
     return null;
   }),
 
+  network: alias('model.allocatedResources.networks.firstObject'),
+  ports: computed('network.reservedPorts.[]', 'network.dynamicPorts.[]', function() {
+    return (this.get('network.reservedPorts') || [])
+      .map(port => ({
+        name: port.Label,
+        port: port.Value,
+        to: port.To,
+        isDynamic: false,
+      }))
+      .concat(
+        (this.get('network.dynamicPorts') || []).map(port => ({
+          name: port.Label,
+          port: port.Value,
+          to: port.To,
+          isDynamic: true,
+        }))
+      )
```

Comment:
Isn't this kind of reshaping generally done in Serializers in nomad? Not sure, maybe check with @DingoEatingFuzz

Reply:
That’s definitely something I’ve seen. In this instance, I’d consider this more of a decoration-for-rendering thing than something that needs to happen in the serialiser. Preprocessing the ports this way lets the two types be combined visually but remain separate in the model. But I really just preserved this from Michael’s preview work so it’s possible he’d prefer these to be merged in the model 😯

Reply:
Understood. I’m unaccustomed to this level of serialiser preprocessing, but I’m also unaccustomed to this style of API. I’ve moved the ports-consolidation into the serialiser, thanks!